app/services/moves/move_type_validator.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Moves
+  class MoveTypeValidator < ActiveModel::Validator
+    attr_reader :record
+
+    def validate(record)
+      @record = record
+      return unless record.move_type.present?
+
+      # Apply more complex validation rules for specific move types
+      validate_police_from_location if includes? %w[video_remand_hearing]
+    end
+
+    private
+
+    def includes?(move_types)
+      move_types.include?(record.move_type.to_s)
+    end
+
+    def human_move_type
+      record.move_type.to_s.humanize(capitalize: false)
+    end
+
+    def validate_police_from_location
+      record.errors.add(:from_location, "must be a police location for #{human_move_type}") unless record.from_location&.police?
```

Comment:
maybe I am wrong, when we add an error, we should specify
- the field,
- the type of error, for example :too_long and 
- the message, 
here the type is missing 🤔

Reply:
Not quite - the second argument can be a string or a symbol to be resolved in the locale file. We have plenty of examples of similar things, but I'll add an explicit test to check the returned error message text.